/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

option java_multiple_files = true;
package org.apache.airavata.mft.credential.stubs.scp;

import "common/CredCommon.proto";

message SCPSecret {
    string secretId = 1;
    string privateKey = 2;
    string publicKey = 3;
    string passphrase = 4;
}

message SCPSecretGetRequest {
    string secretId = 1;
    org.apache.airavata.mft.credential.stubs.common.AuthToken authzToken = 2;
}

message SCPSecretCreateRequest {
    string privateKey = 1;
    string publicKey = 2;
    string passphrase = 3;
    org.apache.airavata.mft.credential.stubs.common.AuthToken authzToken = 5;
}

message SCPSecretUpdateRequest {
    string secretId = 1;
    string privateKey = 2;
    string publicKey = 3;
    string passphrase = 4;
    org.apache.airavata.mft.credential.stubs.common.AuthToken authzToken = 6;
}

message SCPSecretDeleteRequest {
    string secretId = 1;
    org.apache.airavata.mft.credential.stubs.common.AuthToken authzToken = 2;
}

